src/main/java/com/celements/model/access/object/restriction/ObjectQuery.java

```diff
@@ -0,0 +1,75 @@
+package com.celements.model.access.object.restriction;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.celements.model.classes.ClassIdentity;
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+
+@NotThreadSafe
+public class ObjectQuery<O> extends LinkedHashSet<ObjectRestriction<O>> {

```

Comment:
Please do not extend the fundation Collection classes. XWikiContext is a very bad example in this. See "Item 14: Favour composition over inheritance" Is it possible to do here the same with composition instead of inheritance? ObjectQueryBuilder is only using add and addAll which would need to be proxied...